src/mgr/OSDPerfMetricReport.h

```diff
@@ -0,0 +1,29 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#ifndef OSD_PERF_METRIC_REPORT_H_
+#define OSD_PERF_METRIC_REPORT_H_
+#include "include/denc.h"
+//#include "mgr/OSDPerfMetricQuery"
+
+typedef int OSDPerfMetricQuery;  //Temporary
+
+struct OSDPerfMetricReport
+{
+  OSDPerfMetricQuery query; 
+
+  // Gather from OSD::PerfCounters ?
```

Comment:
For the initial version, we can be inefficient so in each report we can encode the expected perf keys (like your comment below) as:

```
struct PerformanceCounter {
  std::string name;
  perfcounter_type_d type;
};

std::vector<PerformanceCounter> performance_counters;
```

We can then encode a map between performance key/group-by (e.g. client global instance id) to a packed bufferlist of the encoded performance counters (encoded/decoded in the order in which they are defined in the previous data structure). See ```MMgrReport::packed``` for an example:

```
std::map<std::string, bufferlist> group_perf;
```

Reply:
Thanks for the feedback - will do accordingly then.

Reply:
Note, in general OSDPerfMetricReport should contain a list of reports (a report per query). You can see one of possible implementations in my PR. This could be improved later though.

Reply:
okay, thanks for the note - I'll have a look for sure